examples/wifi-echo/server/esp32/main/Ble.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/CHIPDeviceLayer.h>
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+using namespace ::chip;
+using namespace ::chip::Ble;
+
+void HandleBleMessageReceived(BLEEndPoint * endPoint, PacketBuffer * buffer)
+{
+    const size_t bufferLen = buffer->DataLength();
+    char msg[bufferLen];
```

Comment:
Is there a max-limit on the Data Length here? The stack usage will vary depending upon this length received.

Reply:
+1, we should definitely avoid large stack-allocated buffers and objects wherever possible.

Reply:
the point of this buffer is logging?  how about only logging the first few bytes of the echo message instead?

Reply:
we also want to sanitize the logging, or is that the job of the device layer?